dbms/src/Storages/KVStore/KVStore.cpp

```diff
@@ -68,6 +68,17 @@ KVStore::KVStore(Context & context)
 {
     // default config about compact-log: rows 40k, bytes 32MB, gap 200.
     LOG_INFO(log, "KVStore inited, eager_raft_log_gc_enabled={}", eager_raft_log_gc_enabled);
+    using namespace std::chrono_literals;
+    monitoring_thread = new std::thread([&]() {
+        while (true)
+        {
+            std::unique_lock l(monitoring_mut);
+            monitoring_cv.wait_for(l, 5000ms, [&]() { return is_terminated; });
+            if (is_terminated)
+                return;
+            recordThreadAllocInfo();
+        }
+    });
```

Comment:
What about register a task to `BackgroundProcessingPool` rather than creating a new thread?

Reply:
I think it brings overhead to `BackgroundProcessingPool` in scheduling.

Reply:
OK